src/AvaloniaEdit/Search/SearchPanel.cs

```diff
@@ -190,29 +195,23 @@ private SearchPanel()
         }
 
         /// <summary>
-        /// Creates a SearchPanel and installs it to the TextEditor's TextArea.
+        /// Creates a SearchPanel and installs it to the TextArea.
         /// </summary>
-        /// <remarks>This is a convenience wrapper.</remarks>
         public static SearchPanel Install(TextEditor editor)
         {
-            if (editor == null)
-                throw new ArgumentNullException(nameof(editor));
-            return Install(editor.TextArea);
-        }
+            var textArea = editor.TextArea;
+            if (textArea == null) throw new ArgumentNullException(nameof(textArea));
 
-        /// <summary>
-        /// Creates a SearchPanel and installs it to the TextArea.
-        /// </summary>
-        public static SearchPanel Install(TextArea textArea)
-        {
-            if (textArea == null)
-                throw new ArgumentNullException(nameof(textArea));
             var panel = new SearchPanel();
+
+            //Subscribe to readonly property
+            editor.GetObservable(TextEditor.IsReadOnlyProperty).Subscribe(x => panel.TextEditorReadOnly = x);
```

Comment:
The property `TextEditorReadOnly` is not an `AvaloniaProperty` so it will not notify the binding When it's updated. I think a simpler approach could be to keep the `Install` methods as they were, add a `TextEditor` property, and bind to `TextEditor.IsReadOnly`.